store/src/main/java/com/onlineStore/coherent/RandomStorePopulator.java

```diff
@@ -1,59 +1,52 @@
 package com.onlineStore.coherent;
 
-import category.Beer;
-import category.Book;
-import category.Category;
-import category.Medicine;
+import com.onlineShop.coherent.category.Category;
+import com.onlineShop.coherent.product.Product;
+import org.reflections.Reflections;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
 
-import com.github.javafaker.Faker;
-import database.DataBase;
-import product.Product;
+import java.util.*;
 
-import java.util.ArrayList;
-import java.util.List;
+import static org.reflections.scanners.Scanners.SubTypes;
 
+@Component
 public class RandomStorePopulator {
-    DataBase dataBase = new DataBase();
-    Faker faker = new Faker();
-    List<Product> productsOfBooks = new ArrayList<>();
-    List<Product> productsOfBeers = new ArrayList<>();
-    List<Product> productsOfMedicines = new ArrayList<>();
-    List<Category> categories = new ArrayList<>();
-    Store store = Store.getStore(categories);
+    private final String URL = "http://localhost:8080";
+    private final Random popular = new Random();
+    private final RestTemplate restTemplate = new RestTemplateBuilder().basicAuthentication("user", "user").build();
 
     public void insertDataOfProducts() {
-        for (int i = 0; i < 3; i++) {
-            dataBase.dataInsertionTemplate(faker.book().title(), faker.number().numberBetween(1, 10), faker.number().numberBetween(1, 1000),"Book");
-            dataBase.dataInsertionTemplate(faker.beer().name(), faker.number().numberBetween(1, 10), faker.number().numberBetween(1, 1000),"Beer");
-            dataBase.dataInsertionTemplate(faker.medical().medicineName(), faker.number().numberBetween(1, 10), faker.number().numberBetween(1, 1000),"Medicine");
+        Reflections reflections = new Reflections("com.onlineShop.coherent.category.categories");
+        Set<Class<?>> subTypes = reflections.get(SubTypes.of(Category.class).asClass());
+        for (Class c : subTypes) {
+            String s = c.getName().replace("com.onlineShop.coherent.category.categories.", "");
+            restTemplate.postForObject(URL + "/category", new Category(s), Category.class);
+        }
+        ResponseEntity<Category[]> responseCategory = restTemplate.getForEntity(URL + "/category", Category[].class);
+        for (Category category : Objects.requireNonNull(responseCategory.getBody())) {
+            for (int i = 0; i < 3; i++) {
+                Product product = new Product.Builder(popular.getProductName(category.getName()), popular.getRate(), popular.getPrice()).build();
+                restTemplate.postForObject(URL + "/product", product, Product.class);
+            }
+            restTemplate.put(URL + "/category/{id}", category, Category.class);
```

Comment:
Could you please explain what this method is doing and why do you need it ?
Also, I am no sure that it works as you expect as you don't provide value instead of {id}

Could you please check status code and response body that service returns for you